File: sdks/nodejs-agent/src/transport/TransportInterface.ts
export interface TransportInterface {
  availabilitySubmit(criteria: Record<string, unknown>): Promise<Record<string, unknown>>;
  availabilityPoll(requestId: string, sinceSeq: number, waitMs: number): Promise<Record<string, unknown>>;
  isLocationSupported(agreementRef: string, locode: string): Promise<boolean>;
  bookingCreate(payload: Record<string, unknown>, idempotencyKey?: string): Promise<Record<string, unknown>>;
  bookingModify(payload: Record<string, unknown>): Promise<Record<string, unknown>>;
  bookingCancel(payload: Record<string, unknown>): Promise<Record<string, unknown>>;
  bookingCheck(supplierBookingRef: string, agreementRef: string, sourceId?: string): Promise<Record<string, unknown>>;
}

File: sdks/java-agent/src/main/java/com/carhire/sdk/transport/TransportInterface.java
package com.carhire.sdk.transport;

import java.util.Map;
import java.util.concurrent.CompletableFuture;

public interface TransportInterface {
    CompletableFuture<Map<String, Object>> availabilitySubmit(Map<String, Object> criteria);
    CompletableFuture<Map<String, Object>> availabilityPoll(String requestId, int sinceSeq, int waitMs);
    CompletableFuture<Boolean> isLocationSupported(String agreementRef, String locode);
    CompletableFuture<Map<String, Object>> bookingCreate(Map<String, Object> payload, String idempotencyKey);
    CompletableFuture<Map<String, Object>> bookingModify(Map<String, Object> payload);
    CompletableFuture<Map<String, Object>> bookingCancel(Map<String, Object> payload);
    CompletableFuture<Map<String, Object>> bookingCheck(String supplierBookingRef, String agreementRef, String sourceId);
}

